crates/esplora/src/blocking_ext.rs

```diff
@@ -139,22 +137,24 @@ impl EsploraExt for esplora_client::BlockingClient {
         })
     }
 
-    fn full_scan<K: Ord + Clone>(
+    fn full_scan<K: Ord + Clone + Send>(
         &self,
-        keychain_spks: BTreeMap<K, impl IntoIterator<Item = (u32, ScriptBuf)>>,
+        request: FullScanRequest<
+            K,
+            impl IntoIterator<IntoIter = impl Iterator<Item = (u32, ScriptBuf)> + Send> + Send + Clone,
```

Comment:
This should just Box an iterator internally. No need for a type parameter. You could even use a concrete SpkIter I guess.